src/code/dialogs/madrid.js

```diff
@@ -66,9 +66,7 @@ const MadridDialog = MultimaxDialog.extend({
       this._setOneDisplay.textContent = wX("NOT_SET");
     }
     L.DomEvent.on(setOneButton, "click", () => {
-      this._portalSetOne = getAllPortalsOnScreen(this._operation).filter(
-        p => p
-      );
+      this._portalSetOne = getAllPortalsOnScreen(this._operation);
       // XXX this is not enough, need to cache them in case IITC purges them
```

Comment:
Is this comment valid? In all the testing I did, it seems safe to not maintain our own cache.

Reply:
right, this comment is not valid anymore since we do use some _cache_.
it's more correct to say we store the portals instead of guid only to do calculation and drawing.